dubbo-dependencies-bom/pom.xml

```diff
@@ -700,6 +700,16 @@
                 <artifactId>grpc-grpclb</artifactId>
                 <version>${grpc.version}</version>
             </dependency>
+            <!-- Groovy, for Embedded Consul. Could be removed once embedded-consul:2.2.1 is released
+             See https://github.com/pszymczyk/embedded-consul/pull/114
```

Comment:
You might wait a little bit with that PR. We plan to release 2.2.1 "soon".

Reply:
I really hope this PR can be merged ASAP, so this issue #7421 can be fixed.

Reply:
@pinxiong I am afraid that this PR won't fix the issue for Mac ARM64. Currently embedded-redis:0.9.0 supports only Linux ARM64. We used Docker+QEMU to build the Linux ARM64 binary - https://github.com/codemonstur/embedded-redis/tree/master/src/main/binaries. I am not sure whether there is a similar way to emulate Mac ARM64.
I am not sure whether @codemonstur has access to Mac M1 machine to build the binary.

Reply:
Thank you for letting me know! If somebody need my help for testing in Mac M1 machine, I'm very happy to do it. @martin-g @codemonstur

Reply:
@pinxiong With PR https://github.com/codemonstur/embedded-redis/pull/3 I've tried to convince @codemonstur to add support for a special system property to ExecutableProvider  to set the path to the redis-server binary but he didn't like this approach.
If this PR was merged then Dubbo/pom.xml could add a `profile` for Mac ARM64 and export this property in Maven Surefire `argline`. Maybe @codemonstur will reconsider it.

BTW once you pass the issue with embedded-redis I expect that you will face similar issue with embedded-consul.

Reply:
I don't have access to a Mac M1 machine. So, no I can't include a binary like that in the library. 

As for the ExecutableProvider; I think the best approach is to override the default behavior by defining your own provider. If you want to load only from a property this code will work:

    ExecutableProvider provider = () -> new File(System.getProperty("my.custom.property"));
    RedisServer redis = RedisServer.newRedisServer()
        .port(DEFAULT_REDIS_PORT)
        .redisExecProvider(provider)
        .build();

I'll be cleaning a few things up and providing some of this stuff out of the box in the next version. Which means you will be able to write this:

    RedisServer redis = RedisServer.newRedisServer()
        .port(DEFAULT_REDIS_PORT)
        // Pick one:
        // will use a property name the library provides
        .executableProvider(ExecutableProvider.newSystemPropertyProvider()) 
        // uses the custom property name
        .executableProvider(ExecutableProvider.newSystemPropertyProvider("my.custom.property"))
        .build();

Reply:
> BTW once you pass the issue with embedded-redis I expect that you will face similar issue with embedded-consul.

@martin-g Thanks for your reminder.

Reply:
@martin-g I have passed the issue with `embedded-redis` when I disabled all test cases in [RedisMetadataReportTest](dubbo-metadata/dubbo-metadata-report-redis/src/test/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReportTest.java). Fortunately, I didn't face similar issue with `embedded-consul`.

Reply:
> You might wait a little bit with merging that PR. We plan to release 2.2.1 "soon".

@szpak Any ETA ?

Reply:
@martin-g  ETA: -15 minutes :-)

https://github.com/pszymczyk/embedded-consul/releases/tag/release%2F2.2.1

Reply:
Thank you, @szpak!
I've upgraded to 2.2.1 with 8825d5200

@AlbumenJ The PR is ready for review and once the CI finishes green for merging!